Content.Shared/StepTrigger/Systems/ShoesRequiredStepTriggerSystem.cs

```diff
@@ -19,18 +18,33 @@ public override void Initialize()
 
     private void OnStepTriggerAttempt(EntityUid uid, ShoesRequiredStepTriggerComponent component, ref StepTriggerAttemptEvent args)
     {
-        if (_tagSystem.HasTag(args.Tripper, "ShoesRequiredStepTriggerImmune"))
+        // checks if the entity itself is the one with the component
+        if (HasComp<ShoesRequiredStepTriggerImmuneComponent>(args.Tripper))
         {
             args.Cancelled = true;
             return;
         }
 
+        // if not, checks the inventory
         if (!TryComp<InventoryComponent>(args.Tripper, out var inventory))
             return;
 
-        if (_inventory.TryGetSlotEntity(args.Tripper, "shoes", out _, inventory))
+        // early exit if no shoe slot exists at all, dionas suffer
+        // remove to give dionas their freedom from glass
+        if (!_inventory.HasSlot(args.Tripper, "shoes", inventory))
+            return;
```

Comment:
this is horribly janky and really just confusing logic. Hardsuits protect from stepping on glass unless you don't have shoe slots in which case they suddenly stop doing that? 

just remove the whole chunk IMO

Reply:
Dionas shall receive their freedom....